src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScore.java

```diff
@@ -0,0 +1,179 @@
+    } else if (stats.averageDaysBetweenReleases < 270) {
+      scoreValue.decrease(2);
+    } else if (stats.averageDaysBetweenReleases < 360) {
+      scoreValue.decrease(3);
+    }
+
+    if (stats.releaseCycleTrend < 0) {
+      scoreValue.decrease(-1 * stats.releaseCycleTrend);
+    } else {
+      scoreValue.increase(stats.releaseCycleTrend);
+    }
+
+    return scoreValue;
+  }
+
+  private VersionStats createVersionStats(Collection<VersionInfo> versionInfos) {
+    IntSummaryStatistics stats = versionInfos.stream()
+        .filter(v -> v.daysDiffToVersionBefore >= 0)
+        .mapToInt(v -> v.daysDiffToVersionBefore)
+        .summaryStatistics();
+
+    int releaseCycleTrend = 0;
+    int lastDays = -1;
+    for (VersionInfo versionInfo : versionInfos) {
+      if (versionInfo.daysDiffToVersionBefore >= 0) {
+        if (lastDays < 0) {
+          lastDays = versionInfo.daysDiffToVersionBefore;
+        } else {
+          if (lastDays > versionInfo.daysDiffToVersionBefore) {
+            releaseCycleTrend--;
+          } else if (lastDays < versionInfo.daysDiffToVersionBefore) {
+            releaseCycleTrend++;
+          }
+        }
+      }
+    }
+
+    double usedReleaseCycleTrend = (double) releaseCycleTrend / stats.getCount();
+    return new VersionStats(usedReleaseCycleTrend, stats.getAverage());
+  }
+
+  private Collection<VersionInfo> createVersionInfos(Collection<ArtifactVersion> artifactVersions) {
+    Collection<VersionInfo> versionInfos = new ArrayList<>();
+    ArtifactVersion beforeVersion = null;
+    Iterator<ArtifactVersion> iterator = artifactVersions.iterator();
+    while (iterator.hasNext()) {
+      boolean latestVersion = false;
+      int daysDiff = -1;
+      ArtifactVersion next = beforeVersion;
+      if (beforeVersion == null) {
+        latestVersion = true;
+        next = iterator.next();
+      }
+      if (iterator.hasNext()) {
+        beforeVersion = iterator.next();
+        daysDiff = (int) DAYS.between(beforeVersion.getReleaseDate(), next.getReleaseDate());
+      }
+      versionInfos.add(new VersionInfo(latestVersion, daysDiff, next));
+    }
+    if (artifactVersions.size() != versionInfos.size()) {
+      // add oldest version
+      versionInfos.add(new VersionInfo(false, -1, beforeVersion));
+    }
+
+    return versionInfos;
+  }
+
+  private static class VersionStats {
+    final double releaseCycleTrend;
+    final double averageDaysBetweenReleases;
+
+    public VersionStats(double releaseCycleTrend, double averageDaysBetweenReleases) {
+      this.releaseCycleTrend = releaseCycleTrend;
+      this.averageDaysBetweenReleases = averageDaysBetweenReleases;
+    }
+
+    @Override
+    public String toString() {
+      return "VersionStats{"
+          + "releaseCycleTrend=" + releaseCycleTrend
+          + ", averageDaysBetweenReleases=" + averageDaysBetweenReleases
+          + '}';
+    }
+  }
+
+  private static class VersionInfo {
+    final boolean latestVersion;
```

Comment:
How do you use this field? Plus, I see that you always sort artifact version before analyzing them. Then, the latest version is always on top (or bottom) of the list.